app/furniture/marketplace/product.rb

```diff
@@ -14,9 +16,14 @@ class Product < Record
 
     has_many :cart_products, inverse_of: :product, dependent: :destroy
     has_many :carts, through: :cart_products, inverse_of: :products
+    after_discard do
+      # This is redonkulus, and due to the inappropriate polymorphism where
+      # carts/orders are the same table.
+      cart_products.where(cart: carts).destroy_all
+    end
 
     has_many :ordered_products, inverse_of: :product, dependent: :destroy
-    has_many :orders, through: :ordered_products, inverse_of: :products
+    has_many :orders, -> { checked_out }, through: :ordered_products, inverse_of: :products
```

Comment:
For my edification, how does "checked out" differ from "paid"?

Reply:
`paid` is probably fine to use instead at this time; but as we think about what states an Order can flow through (`placed`, `paid`, `preparing`, `ready_for_pickup`, `in_transit`, `delivered`, `reviewed`) 

Tho to be honest, all this goes away once I get the `Cart` and `Order` into different tables. That's just a project for Spoons Having Zee

Reply:
It sounds like what you're saying is that they are the same, which makes sense since it only looks like nominal mapping to me that happens at https://github.com/zinc-collective/convene/blob/main/app/furniture/marketplace/order.rb#L4.

Reply:
Ehh, they're not exactly the same. `where.not(status: :pre_checkout)` will return records where status is `preparing`, `ready_for_pickup`, etc, whereas the `paid` scope will only return records `where(status: :paid)`.

Reply:
Right, what I was trying to understand is if they could be effectively considered equivalent _states_ in the current code since the `status` enum only declares `pre_checkout` and `paid`. But I get what you mean here.

Reply:
I would say they're equivalent _for now_ rather than that they're the same; but objective equality is hard enough in deterministic computational systems; much less ones subjected to the chaos of temporality 🤣